### DISCUSSION TITLE
Insulin Input

### DISCUSSION THREAD

**Discussion by robertherber** [nielsmaerten/gluco-check#129] [Ideas]
It'd be great to be able to tell Nightscout when I'm taking insulin and how many units.

**Comment by nielsmaerten** [comment]
I agree this would be a great feature, but I'm afraid Gluco Check will likely never support this. Not because it can't be done... but because Google won't allow it.

In short, we're an app in the "Health and Fitness" category. Per Google's policies, we **can only read information back to you for the purpose of activity tracking and monitoring**. 

If Gluco Check could enter new boluses into Nightscout, that would constitute logging potentially medical actions. For those of us using closed loops, it could even be dangerous: what if the Assistant mishears what you're saying and enters an incorrect value?

So for now, Gluco Check will remain a read-only app. At least until voice recognition becomes 100% flawless 😉

**Reply by utzaki** [reply]
@robertherber I agree and was close to adding the same idea here in a new thread. Some time ago, I implemented the option of adding insulin and/or carbs by using Assistant -> Ifttt (Webhook) -> Nightscout. This is working great and could be a good feature for Gluco Check. But I understand Niels, that this may be a huge risk and therefore violate Google policies. So I will keep using my Ifttt method, which works well.
